.esh/commands/update-locales.py

```
@@ -1457,15 +1444,53 @@ def generate_ts_type_recursive(data, indent_level):
-			if isinstance(value, (OrderedDict, dict)):
+			if isinstance(value, list):
+				parts.append(f"{indent}{ts_key}: {generate_ts_array_type(value, indent_level)}")
+			elif isinstance(value, (OrderedDict, dict)):
 				parts.append(f"{indent}{ts_key}: {generate_ts_type_recursive(value, indent_level + 1)}")
 			elif isinstance(value, str):
 				parts.append(f"{indent}{ts_key}: string")
 		parts.append("\t" * (indent_level - 1) + "}")
 		return "\n".join(parts)
 
-	locale_data_type_str = generate_ts_type_recursive(ref_data, indent_level=1)
+	def generate_ts_array_type(arr, indent_level):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `generate_ts_array_type`

Add return type annotation: `str`

(ANN202)